Metro-Records/la-metro-councilmatic
# A11y Improvements - Operable

## Overview

This branch aims to improve on the "operable" category of WCAG standards for the app. Each of the areas of improvement are listed in the connected issue.

Most notable changes are that maps and the search radio buttons now maintain visible focus when tabbed over, and each link that opens a page in a new tab now announces that to screen reader users in order to [prevent confusion for those users](https://www.w3.org/TR/WCAG20-TECHS/G201.html). Also, a footer has been added with a link to the metro login page to satisfy "2.4.5 Multiple Ways".

- Connects #1022 

### Notes

The original issues for each category are listed in #1022

- 2.1.1 Keyboard:
  - I believe the board members map has a sufficient alternative in the form of the listing table adjacent to it. And the map on each member's detail page has a `figcaption` (previously a header) that serves as a label/descriptor.
- 2.4.7 Focus Visible:
  - TIL about the `focus-within` css selector. It was a huge help to get visible focus onto the label for the search options instead of just on the invisible radio button itself.
- Extra:
  - My original plan for the new tab warnings was to add an alt tag to existing external link icons that occasionally appear next to links, then adding that icon to links that didn't have it. However there were a number of links where adding that would have looked messy, like those where the link was a part of a paragraph. So instead, I added an `aria-label` to each one of those links for screen readers.

## Testing Instructions

 * Get to the site
 * Asses if the points covered in the issue have been properly addressed.
   * Use lighthouse or another a11y tool to find out if headings are in the proper order.
   * Tab through maps and links to make sure visible focus is maintained at all times.
   * Use a screen reader like VoiceOver to ensure that new tab warnings, map descriptions, etc. are being announced properly.
